clippy_utils/src/consts.rs

```diff
@@ -788,6 +834,40 @@ pub fn mir_to_const<'tcx>(lcx: &LateContext<'tcx>, result: mir::Const<'tcx>) ->
     }
 }
 
+fn mir_is_empty<'tcx>(lcx: &LateContext<'tcx>, result: mir::Const<'tcx>) -> Option<bool> {
```

Comment:
It looks like most `is_empty` functions in rustc's std are already static. Could we maybe make our lives easier and throw the whole `<recv>.is_empty()` expression into the constant evaluation? That seems easier than this entire thing.

(I haven't worked with const too much, so excuse me, if this suggestion is invalid)

Reply:
Yes, probably, if those methods were `const` they would not trigger a warning but the use of their result might.